### ISSUE TITLE
Investigate causal modeling + ML ensembles for short-horizon risk & alpha (arXiv 2510.22348)

### ISSUE THREAD

**Issue by JohnGavin** [JohnGavin/historical#122]
## Source

**Paper:** [Causal and Predictive Modeling of Short-Horizon Market Risk and Systematic Alpha Generation Using Hybrid Machine Learning Ensembles](https://arxiv.org/abs/2510.22348)  
**Author:** Aryan Ranjan  
**Published:** October 25, 2025

---

## Abstract Summary

**Framework:** Predict **5-day S&P 500 drawdowns** using hybrid ML ensemble (neural networks + tree-based voting models)

**Data:** Cross-asset features spanning equities, fixed income, FX, commodities, volatility

**Results:**
- **Sharpe ratio:** 2.51
- **Annualized CAPM alpha:** +0.28
- **Interpretability:** Feature attribution reveals macro + microstructural drivers of crash vs non-crash regimes

---

### Short-Horizon Focus (5-Day Prediction)

**Contrast with traditional approaches:**
- Traditional: monthly/quarterly rebalancing, long-horizon risk models
- This paper: **5-day crash prediction**, daily trading decisions

**Why 5 days?**
- Matches typical institutional risk management horizon
- Captures intra-week volatility clustering
- Avoids overfitting to daily noise

### Causal + Predictive Hybrid

**Causal component:**
- Identifies **drivers** of crash regimes (interpretability)
- Uses feature attribution methods (SHAP, permutation importance)

**Predictive component:**
- Neural networks for non-linear patterns
- Tree-based models for robust voting
- Ensemble combines both for final signal

**Advantage:** Not just predicting crashes, but understanding **why** they happen (macro vs microstructure).

---

### Current Gap: No Short-Horizon Risk Model

Our existing risk management:
- **VIX overlay** (#105, Tier 2): daily VIX threshold, but binary signal (on/off)
- **Avoid Worst** (\`plan_avoid_worst.R\`): VIX-based crisis protection
- **VVIX analysis** (#105): volatility-of-volatility regimes

**Missing:** Probabilistic 5-day drawdown forecast with continuous sizing.

### Comparison to Existing Work

| Approach | Horizon | Features | Model | Output |
|---|---|---|---|---|
| **VIX overlay** (ours) | Daily | VIX level | Threshold rule | Binary (in/out) |
| **VVIX regimes** (ours) | Daily | VIX + VVIX | K-means clustering | 4 regimes |
| **This paper** | 5-day | Cross-asset (100+ features) | NN + tree ensemble | Probability(crash) |
| **StockGPT** (#120) | Daily | Raw returns | Transformer | Predicted return |

**Synergy:** This paper's crash probability could **modulate** existing strategies (reduce size when P(crash) > threshold).

---

### Phase 1: Understand Architecture

**Tasks:**
- [ ] Read full paper (arXiv 2510.22348)
- [ ] Document model components:
  - Neural network architecture (layers, activation, regularization)
  - Tree-based voting (XGBoost? Random Forest? Gradient boosting?)
  - Ensemble weighting (stacking? averaging?)
  - Training objective (binary classification? regression on drawdown magnitude?)
- [ ] Identify feature set:
  - Cross-asset: which assets? (SPY, TLT, GLD, DBC, VIX confirmed?)
  - Technical: what indicators? (momentum, volatility, volume?)
  - Macro: what signals? (rates, credit spreads, sentiment?)
- [ ] Feature attribution methods:
  - SHAP values for driver identification
  - Permutation importance for robustness check

**Deliverable:** \`knowledge/wiki/causal-crash-prediction.md\`

---

### Phase 2: Replication Feasibility

**Option A: Full Replication (High Effort)**
- Train hybrid ensemble from scratch on S&P 500 5-day returns (2000-present)
- Requires: Python (PyTorch/TF + XGBoost), GPU, extensive hyperparameter tuning
- Timeline: 2-3 weeks
- Risk: Many undocumented implementation details

**Option B: Simplified Version (Medium Effort)**
- Use pre-trained ensemble libraries (sklearn stacking)
- Train on smaller feature set (10-20 most important from paper)
- Timeline: 1 week
- Risk: May not achieve Sharpe 2.5, but can validate concept

**Option C: Feature Extraction Only (Low Effort)**
- Extract the top 10 features identified by SHAP
- Add to existing VIX overlay as additional conditions
- Timeline: 2-3 days
- Risk: Lose ensemble benefits, but quick implementation

**Recommendation:** Start with **Option C** (feature extraction), then decide on full replication.

---

### Phase 3: Integration with Existing Strategies

**Use crash probability to modulate position sizing:**

\`\`\`r
# R/plan_crash_risk.R
tar_target(crash_probability, predict_5day_crash(cross_asset_features))

tar_target(modulated_strategy_returns, {
  # Reduce size when P(crash) > 0.3
  size_scalar <- if_else(crash_probability > 0.3, 0.5, 1.0)
  strategy_returns * size_scalar
})
\`\`\`

**Integration points:**

| Strategy | Current Risk Mgmt | + Crash Probability |
|---|---|---|
| **LTR Momentum** | None | Reduce size when P(crash) high |
| **DRIF** | None | Scale elastic net bets by crash risk |
| **Avoid Worst** | VIX > 30 = 0% equity | Continuous scaling: P(crash) → equity % |
| **Factormax** | Sharpe weighting | Add crash probability as 6th factor |

**Tasks:**
- [ ] Test crash-modulated sizing on LTR, DRIF, Factormax
- [ ] Compare to VIX-only overlay (#105)
- [ ] Measure impact on Sharpe, max drawdown, tail risk

---

### Phase 4: Causal Driver Analysis

**Key question:** What drives 5-day crashes? (from paper's SHAP analysis)

**Expected drivers (to verify):**
- **Macro:** credit spreads widening, yield curve inversion, inflation surprises
- **Microstructure:** volume surges, put/call ratio spikes, bid-ask spread widening
- **Cross-asset:** TLT/SPY correlation flip, VIX spike, commodity crash

**Tasks:**
- [ ] Extract top 10 SHAP features from paper
- [ ] Verify we have these features in our data:
  - Credit spreads: TED spread, HY-IG spread (missing — add to weekly poll?)
  - Put/call ratio: CBOE data (missing)
  - Volume: already have in OHLCV
- [ ] Build dashboard: \`crash-drivers.qmd\`
  - Current values of top 10 features
  - Historical crash regime periods
  - Feature contribution to current crash probability

---

### Deliverables:
1. **Wiki page:** \`knowledge/wiki/causal-crash-prediction.md\`
   - Paper summary, architecture, results
   - Top 10 crash drivers (SHAP features)
   - Comparison to VIX overlay

2. **Functions:** \`R/crash_risk.R\` (if Phase 3 proceeds)
   - \`predict_5day_crash()\`: simplified ensemble
   - \`modulate_by_crash_risk()\`: position sizing adjustment
   - \`crash_driver_dashboard()\`: feature contribution plot

3. **Targets:** \`R/plan_crash_risk.R\` (if Phase 3 proceeds)
   - \`cross_asset_features\`: feature matrix
   - \`crash_probability\`: 5-day P(crash)
   - \`crash_modulated_returns\`: size-adjusted strategies

4. **Vignette:** \`crash-risk-modeling.qmd\` (if full replication)
   - Model architecture and training
   - Feature importance analysis
   - Strategy integration results

---

## Related Issues

- **#105 (VIX overlay):** This is a more sophisticated version
- **#119 (Volatility spikes):** Crash probability complements spike detection
- **#120 (StockGPT):** Both use ML, but StockGPT predicts returns, this predicts crashes
- **#117 (DRIF):** Crash probability could modulate DRIF position sizes

---

## Timeline

- **Phase 1:** 3-4 hours (read paper, document architecture)
- **Phase 2:** 2-3 days (feature extraction) OR 1 week (simplified) OR 2-3 weeks (full)
- **Phase 3:** 4-5 hours (integration with existing strategies)
- **Phase 4:** 3-4 hours (causal driver analysis)

**Minimum (Phase 1+2C+3):** 1-2 days  
**Maximum (Phase 1+2A+3+4):** 3-4 weeks

---

## Priority

**Medium-High** — Short-horizon risk prediction is valuable, especially given:
- Sharpe 2.5 is exceptional (if replicable)
- Crash modulation could improve all existing strategies
- Feature extraction (Phase 2C) is low-effort, high-value

**Decision point:** After Phase 1, decide whether Sharpe 2.5 result is credible enough to justify replication effort.

---

## Sources

- [Causal and Predictive Modeling of Short-Horizon Market Risk (arXiv 2510.22348)](https://arxiv.org/abs/2510.22348)
- Aryan Ranjan (2025)
